### PR TITLE
fix: take feature channels in consideration

### PR DESCRIPTION
**Pull request by nichmor**
fixes: https://github.com/prefix-dev/pixi/issues/1988

### CODE REVIEW

**Comment by nichmor** [comment]
> Can you also add a test?

yes! will add it

File: src/project/mod.rs [hunk 1]
```diff
@@ -954,4 +968,38 @@ mod tests {
             &MappingLocation::Path(PathBuf::from("mapping.json"))
         );
     }
+
+    #[test]
+    fn test_mapping_check_also_feature_channels() {
```
Comment (tdejager) [inline src/project/mod.rs:973]:
```suggestion
    fn test_mapping_ensure_feature_channels_also_checked() {
```

File: src/project/snapshots/pixi__project__tests__mapping_check_also_feature_channels.snap [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+source: src/project/mod.rs
+expression: project.pypi_name_mapping_source().unwrap_err()
+---
+Defined conda-pypi-map channel: non-existing-channel is missing from the channels, which are: conda-forge, pytorch
```
Comment (tdejager) [inline src/project/snapshots/pixi__project__tests__mapping_check_also_feature_channels.snap:5]: I think we can improve this message with:

```suggestion
conda-pypi-map is defined: the `non-existing-channel` is missing from the channels array, which currently are: conda-forge, pytorch
```

**Review by tdejager — CHANGES_REQUESTED** [review]

**Review by tdejager — CHANGES_REQUESTED** [review]
Would you mind adding a custom `thiserror` and using miette a bit higher up the chain, this would require some refactoring though.

**Review by baszalmstra — APPROVED** [review]

**Review by tdejager — APPROVED** [review]

**Pull request merged**